### ISSUE TITLE
Curved Bone Kookaburra - The user has no control over the crosschain transaction because `delegate` is `address(0)`

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-06-tokensoft-judging#76]
Curved Bone Kookaburra

medium

# The user has no control over the crosschain transaction because `delegate` is `address(0)`

## Summary

The user has no control over the crosschain transaction because `delegate` is `address(0)`

## Vulnerability Detail

In `Satellite.sol` we have `initiateClaim()` function. This fucntion initiates crosschain claim by msg.sender. `initiateClaim()` call `connext.xcall`:

```solidity
87:    // Send claim to distributor via crosschain call
88:    bytes32 transferId = connext.xcall{ value: msg.value }(  
89:      _distributorDomain, // destination domain              
90:      address(distributor), // to
91:      address(0), // asset    //@audit - missing delegate
92:      address(0), // delegate, only required for self-execution + slippage
93:      0, // total
94:      0, // slippage
95:      abi.encodePacked(msg.sender, _domain, total, proof) 
96:    );
```

On line 92 we notice that in place of `delegate` we have `address(0)` assuming you won't have to because there's no slippage. But `delegate` is important not only because of slippage.

According to connext docs `delegate`  is:

> An address on destination domain that has rights to update slippage tolerance, retry transactions, or revert back to origin in the event that a transaction fails at the destination.

[Reference](https://docs.connext.network/developers/reference/contracts/calls#functions)

Left at `address(0)`, absolutely no one has control of the transaction and according to the documentation it is entirely possible that the transaction remains stuck or some event happens and transaction fails at the destination. 

## Impact

When a transaction remains stuck or some event happens and the transaction fails at the destination we have no way to respond.

## Code Snippet

https://docs.connext.network/developers/reference/contracts/calls#functions
https://github.com/sherlock-audit/2023-06-tokensoft/blob/main/contracts/contracts/claim/Satellite.sol#L88-L96

## Tool used

Manual Review

## Recommendation

Replace `address(0)` with `msg.sender` or `address(distributor)` to have some control over the transaction in case it gets stuck or you need to repeat it.

**Comment by cvetanovv** [comment]
Escalate

I think this is a valid medium issue. If `delegate` is 0, this means no one has control of the transaction and according to the documentation, it is entirely possible that the transaction remains stuck or some event happens and the transaction fails at the destination.

https://docs.connext.network/developers/reference/contracts/calls#functions

**Comment by sherlock-admin2** [comment]
> Escalate
> 
> I think this is a valid medium issue. If `delegate` is 0, this means no one has control of the transaction and according to the documentation, it is entirely possible that the transaction remains stuck or some event happens and the transaction fails at the destination.
> 
> https://docs.connext.network/developers/reference/contracts/calls#functions
> 

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by Shogoki** [comment]
I do not think this is a Medium issue here. Because there are no tokens transferred using the `xcall`. It is only. another function that is called on the CrossChainDistributor on the target domain.
All important state checks and changes are done inside CrossChainDistributor on the target domain/chain. 
So, if the xcall fails or get´s stuck there are no funds at loss, but anybody can just call the the function again or create their own xCall to initiate the claim.

**Comment by hrishibhat** [comment]
@cvetanovv

**Comment by cr-walker** [comment]
I agree with @Shogoki's assessment: claims can't get stuck this way because the Satellite contracts are stateless. If the transaction fails on the destination chain we can just call `initiateClaim()` again.

**Comment by hrishibhat** [comment]
Result:
Low
Unique
Considering this low issue based on the above comments

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [cvetanovv](https://github.com/sherlock-audit/2023-06-tokensoft-judging/issues/76/#issuecomment-1669334919): rejected
